Improve rosetta docs (#727)
o1-labs/docs2: docs/exchange-operators/rosetta/samples/send-transactions.mdx

---
title: Sending transactions

> description: Learn how to send transactions using a keypair

Reviewer: ```suggestion
description: Learn how to send transactions using a key pair
```

---

:::info
This flow is also described in the [Construction API Overview](https://docs.cloud.coinbase.com/rosetta/docs/construction-api-overview) article in the official Rosetta documentation.
:::

> The steps needed to send payment in MINA token are following:

Reviewer: ```suggestion
The steps to send a payment in MINA tokens:
```

> 1. Given a keypair, derive the account indentifier using `/construction/derive` endpoint

Reviewer: I won't mark the rest instances, but be sure to use two words for "key pair' 
https://github.com/o1-labs/docs2/wiki/Word-list#key-pair

Author: @barriebyron done

2. Call `/construction/preprocess` and `/construction/metadata` to construct parameters for `/construction/payloads` request
3. Create an unsigned transaction blob using `construction/payloads` endpoint
4. Call `construction/parse` (optional) to check if the unsigned transaction does what you expect
5. Use detached signer to sign the transaction
6. Call `construction/combine` to generate signed blob to be sent via `/construction/submit` endpoint
7. Call `construction/parse` again (optional) to confirm correctness of the signed transaction
8. Get a future transaction hash using `/construction/hash` enpoint
9. Submit the signed transaction blob via `/construction/submit` endpoint

> For ease of readability, our sample implementation skips sanity checks (steps 4 and 7).
> We have also combined steps 2 and 3 in a single `tx_payloads` function call.

Reviewer: ```suggestion
For ease of readability, this sample implementation skips the sanity checks (steps 4 and 7) and combines steps 2 and 3 in a single `tx_payloads` function call.
```
